Model/Filter/FilterBuilder.php

```diff
@@ -0,0 +1,196 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Model\Filter;
+
+use Magento\Catalog\Model\Layer\Filter\Item;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\Store;
+use Nosto\Operation\Recommendation\IncludeFilters;
+use Nosto\Operation\Recommendation\ExcludeFilters;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+
+class FilterBuilder
+{
+    /** @var IncludeFilters */
+    private $includeFilters;
+
+    /** @var ExcludeFilters */
+    private $excludeFilters;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var string */
+    private $brand;
+
+    /** @var NostoLogger */
+    private $logger;
+
+    /**
+     * FilterBuilder constructor.
+     * @param IncludeFilters $includeFilters
+     * @param ExcludeFilters $excludeFilters
+     * @param NostoHelperData $nostoHelperData
+     * @param NostoLogger $logger
+     */
+    public function __construct(
+        IncludeFilters $includeFilters,
+        ExcludeFilters $excludeFilters,
+        NostoHelperData $nostoHelperData,
+        NostoLogger $logger
+    ) {
+        $this->includeFilters = $includeFilters;
+        $this->excludeFilters = $excludeFilters;
+        $this->nostoHelperData = $nostoHelperData;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Store $store
+     */
+    public function init(Store $store): void
```

Comment:
I don't think this is compatible with PHP 7.0. Was introduced in 7.1:
https://www.php.net/manual/en/migration71.new-features.php